planner/core/testdata/partition_pruner_out.json

```diff
@@ -1646,7 +1646,7 @@
         ],
         "IndexPlan": [
           "Sort 199.80 root  test_partition_1.t1.a",
-          "└─IndexJoin 199.80 root  inner join, inner:IndexReader, outer key:test_partition_1.t2.b, inner key:test_partition_1.t1.a, equal cond:eq(test_partition_1.t2.b, test_partition_1.t1.a)",
+          "└─IndexHashJoin 199.80 root  inner join, inner:IndexReader, outer key:test_partition_1.t2.b, inner key:test_partition_1.t1.a, equal cond:eq(test_partition_1.t2.b, test_partition_1.t1.a)",
```

Comment:
This change can be ignored since `IndexJoin` and `IndexHashJoin` always have the same cost in this case, so it's ok to choose any one of them.

Before this PR:
```
[Sort_12 199.80 9216.51 root  test_partition_1.t1.a]
[└─IndexJoin_19 199.80 4359.68 root  inner join, inner:IndexReader_18, outer key:test_partition_1.t2.b, inner key:test_partition_1.t1.a, equal cond:eq(test_partition_1.t2.b, test_partition_1.t1.a)]
[  ├─HashAgg_28(Build) 159.84 1177.19 root  group by:test_partition_1.t2.b, funcs:firstrow(test_partition_1.t2.b)->test_partition_1.t2.b]
[  │ └─IndexReader_29 159.84 1048.29 root partition:p0 index:HashAgg_24]
[  │   └─HashAgg_24 159.84 0.00 cop[tikv]  group by:test_partition_1.t2.b, ]
[  │     └─IndexRangeScan_27 199.80 13786.20 cop[tikv] table:t2, index:a(a, b, id) range:[1 -inf,1 +inf], [2 -inf,2 +inf], keep order:false, stats:pseudo]
[  └─IndexReader_18(Probe) 1.25 9.37 root partition:p0,p1 index:Selection_17]
[    └─Selection_17 1.25 0.00 cop[tikv]  not(isnull(test_partition_1.t1.a))]
[      └─IndexRangeScan_16 1.25 0.00 cop[tikv] table:t1, index:a(a, b, id) range: decided by [eq(test_partition_1.t1.a, test_partition_1.t2.b)], keep order:false, stats:pseudo]

[Sort_34 199.80 8940.75 root  test_partition_1.t1.a]
[└─IndexHashJoin_33 199.80 4359.68 root  inner join, inner:IndexReader_30, outer key:test_partition_1.t2.b, inner key:test_partition_1.t1.a, equal cond:eq(test_partition_1.t2.b, test_partition_1.t1.a)]
[  ├─HashAgg_23(Build) 159.84 1177.19 root  group by:test_partition_1.t2.b, funcs:firstrow(test_partition_1.t2.b)->test_partition_1.t2.b]
[  │ └─IndexReader_24 159.84 1048.29 root partition:p0 index:HashAgg_19]
[  │   └─HashAgg_19 159.84 0.00 cop[tikv]  group by:test_partition_1.t2.b, ]
[  │     └─IndexRangeScan_22 199.80 13786.20 cop[tikv] table:t2, index:a(a, b, id) range:[1 -inf,1 +inf], [2 -inf,2 +inf], keep order:false, stats:pseudo]
[  └─IndexReader_30(Probe) 1.25 9.37 root partition:p0,p1 index:Selection_29]
[    └─Selection_29 1.25 0.00 cop[tikv]  not(isnull(test_partition_1.t1.a))]
[      └─IndexRangeScan_28 1.25 0.00 cop[tikv] table:t1, index:a(a, b, id) range: decided by [eq(test_partition_1.t1.a, test_partition_1.t2.b)], keep order:false, stats:pseudo]
```

After this PR:
```
[Sort_10 199.80 8992.47 root  test_partition_1.t1.a]
[└─IndexJoin_16 199.80 4273.53 root  inner join, inner:IndexReader_15, outer key:test_partition_1.t2.b, inner key:test_partition_1.t1.a, equal cond:eq(test_partition_1.t2.b, test_partition_1.t1.a)]
[  ├─HashAgg_23(Build) 159.84 1091.04 root  group by:test_partition_1.t2.b, funcs:firstrow(test_partition_1.t2.b)->test_partition_1.t2.b]
[  │ └─IndexReader_24 159.84 1048.29 root partition:p0 index:HashAgg_19]
[  │   └─HashAgg_19 159.84 0.00 cop[tikv]  group by:test_partition_1.t2.b, ]
[  │     └─IndexRangeScan_22 199.80 13786.20 cop[tikv] table:t2, index:a(a, b, id) range:[1 -inf,1 +inf], [2 -inf,2 +inf], keep order:false, stats:pseudo]
[  └─IndexReader_15(Probe) 1.25 9.37 root partition:p0,p1 index:Selection_14]
[    └─Selection_14 1.25 0.00 cop[tikv]  not(isnull(test_partition_1.t1.a))]
[      └─IndexRangeScan_13 1.25 0.00 cop[tikv] table:t1, index:a(a, b, id) range: decided by [eq(test_partition_1.t1.a, test_partition_1.t2.b)], keep order:false, stats:pseudo]

[Sort_34 199.80 8854.59 root  test_partition_1.t1.a]
[└─IndexHashJoin_33 199.80 4273.53 root  inner join, inner:IndexReader_30, outer key:test_partition_1.t2.b, inner key:test_partition_1.t1.a, equal cond:eq(test_partition_1.t2.b, test_partition_1.t1.a)]
[  ├─HashAgg_23(Build) 159.84 1091.04 root  group by:test_partition_1.t2.b, funcs:firstrow(test_partition_1.t2.b)->test_partition_1.t2.b]
[  │ └─IndexReader_24 159.84 1048.29 root partition:p0 index:HashAgg_19]
[  │   └─HashAgg_19 159.84 0.00 cop[tikv]  group by:test_partition_1.t2.b, ]
[  │     └─IndexRangeScan_22 199.80 13786.20 cop[tikv] table:t2, index:a(a, b, id) range:[1 -inf,1 +inf], [2 -inf,2 +inf], keep order:false, stats:pseudo]
[  └─IndexReader_30(Probe) 1.25 9.37 root partition:p0,p1 index:Selection_29]
[    └─Selection_29 1.25 0.00 cop[tikv]  not(isnull(test_partition_1.t1.a))]
[      └─IndexRangeScan_28 1.25 0.00 cop[tikv] table:t1, index:a(a, b, id) range: decided by [eq(test_partition_1.t1.a, test_partition_1.t2.b)], keep order:false, stats:pseudo]
```